plugins/modules/ec2_imagebuilder_component.py

```diff
@@ -0,0 +1,399 @@
+from ansible_collections.amazon.aws.plugins.module_utils.aws.core import AnsibleAWSModule
+
+try:
+    import botocore
+except ImportError:
+    pass  # caught by AnsibleAWSModule
+
+
+# Non-ansible imports
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass
+
+
+def construct_component_arn(region, account_id, name, semantic_version):
+    return "arn:aws:imagebuilder:%s:%s:component/%s/%s/1" % (
+        region, account_id, name.replace('_', '-'), semantic_version)
+
+
+def construct_create_params(module):
+    params = dict()
+
+    # required params
+    params['name'] = module.params.get('name')
+    params['platform'] = module.params.get('platform')
+    params['semantic_version'] = module.params.get('semantic_version')
+
+    # optional params
+    optional_params_keys = [
+        'change_description',
+        'data',
+        'description',
+        'kms_key_id',
+        'uri',
+        'tags']
+
+    for k in optional_params_keys:
+        if module.params.get(k) is not None:
+            params[k] = module.params.get(k)
+
+    # ensure params are in camel case for compatibility with EC2 Image Builder API
+    params = snake_dict_to_camel_dict(params)
+
+    return params
+
+
+def create_or_tag_component(connection, module, arn, current_component_params):
+    params = construct_create_params(module)
+
+    changed = False
+
+    # attempt to create component if it does not exist or its parameters have changed
+    if not current_component_params or has_component_params_changed(
+            params, current_component_params):
+        connection.create_component(**params)
+        changed = True
+
+    # update tags if component exists and tags have changed
+    if current_component_params and have_component_tags_changed(params, current_component_params):
+        update_component_tags(connection, arn, params, current_component_params)
+        changed = True
+
+    # if a change has been made, refetch
+    if changed:
+        current_component_params = get_component(connection, module, arn)
+
+    module.exit_json(changed=changed, component=camel_dict_to_snake_dict(current_component_params))
+
+
+def delete_component(connection, module, arn, component):
+    """
+    Delete an EC2 Image Builder component if it exists
+
+    :param connection: AWS boto3 imagebuilder connection
+    :param module: Ansible module
+    :param component: a dict of EC2 Image Builder component representing current state of resource, or None
+    :return: True if component was deleted else False
+    """
+
+    changed = False
+
+    if component:
+        try:
+            connection.delete_component(
+                componentBuildVersionArn=arn)
+            changed = True
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e)
+
+    module.exit_json(changed=changed, component=dict(arn=arn))
+
+
+def get_aws_account_id(module):
```

Comment:
There's no guarantee that get_caller_identity would actually return what we want here.  Given that the boto3 APIs seem to all be based on using an ARN and you can't get by name, I'd recommend
- Support passing an ARN instead of just the name (you can pass `required_one_of=["arn", "name"]` as an argument when creating the module)
- Wrapping this in a try/except block and only call it when passed a name instead of an ARN.
- If the block fails then recommend using the resource ARN instead of the name.